### PR TITLE
CRM-16356

### PR DESCRIPTION
**Pull request by caracolmzd**
Add UNIQUE to name field of civicrm_setting;
Changes to update/install for same.
Enable use of api filters param to setting.get;

---
- [CRM-16356: Settings group_name cannot be specified in api.setting.create.](https://issues.civicrm.org/jira/browse/CRM-16356)

### CODE REVIEW

**Comment by ginkgomzd** [comment]
Not ready to merge. Need to run tests, and probably write one test. 
Had some review by @eileenmcnaughton already. Could use some more from @totten .
@GinkgoFJG

File: CRM/Upgrade/Incremental/php/FourSix.php [hunk 1]
```diff
@@ -220,4 +223,19 @@ public static function fixCaseLog(CRM_Queue_TaskContext $ctx) {
     return TRUE;
   }
 
+  public function upgrade_4_6_3($rev) {
```
Comment (ginkgomzd) [inline CRM/Upgrade/Incremental/php/FourSix.php]: @totten Are the docs out of date? http://wiki.civicrm.org/confluence/display/CRMDOC/Upgrade+Reference#UpgradeReference-IncrementalSQLFiles
Is this function no longer required?

File: CRM/Upgrade/Incremental/php/FourSix.php [hunk 2]
```diff
@@ -56,6 +56,9 @@ public function verifyPreDBstate(&$errors) {
    * @param null $currentVer
    */
   public function setPreUpgradeMessage(&$preUpgradeMessage, $rev, $currentVer = NULL) {
+    if (!$this->validateUniqueSettingNames()) {
```
Comment (ginkgomzd) [inline CRM/Upgrade/Incremental/php/FourSix.php]: @totten Do you think this should be added to the docs as the recommended place for DB pre-checks for minor releases? Docs say that verifyPreDBstate() is not called for minor releases.

**Comment by ginkgomzd** [comment]
Looks like the unique constraint is wrong. Uniqueness needs to be of name and domain_id.

**Comment by eileenmcnaughton** [comment]
Good point!

**Comment by eileenmcnaughton** [comment]
In my head I thought this was merged - how do you feel about rebasing it once 4.6.6 is out & trying to get it into 4.6.7

**Comment by davecivicrm** [comment]
Due to major changes in settings for 4.7, pretty sure this is obsolete.

**Pull request closed without merging**